src/main/java/application/rest/v1/KAppNavEndpoint.java

```diff
@@ -314,6 +320,196 @@ public static String getComponentNamespace(JsonObject component) {
+            if (inputs == null) {
+                if (Logger.isExitEnabled()) {
+                    Logger.log(className, methodName, Logger.LogType.EXIT, "Inputs not found");
+                }
+                return;
+            }
+            if (Logger.isDebugEnabled()) {
+                Logger.log(className, methodName, Logger.LogType.DEBUG, "inputs found");
+            }
+            Set<Entry<String, JsonElement>> inputEntries = inputs.entrySet();
+            for (Map.Entry<String, JsonElement> inputEntry : inputEntries) {
+                try {
+                    if (Logger.isDebugEnabled()) {
+                        Logger.log(className, methodName, Logger.LogType.DEBUG, "input found: " + inputEntry.getKey());
+                    }
+                    JsonObject fields = inputEntry.getValue().getAsJsonObject().getAsJsonObject("fields");
+                    if (fields != null) {
+                        if (Logger.isDebugEnabled()) {
+                            Logger.log(className, methodName, Logger.LogType.DEBUG, "fields found");
+                        }
+                        Set<Entry<String, JsonElement>> fieldEntries = fields.entrySet();
+                        for (Map.Entry<String, JsonElement> fieldEntry : fieldEntries) {
+                            try {
+                                if (Logger.isDebugEnabled()) {
+                                    Logger.log(className, methodName, Logger.LogType.DEBUG,
+                                            "field found: " + fieldEntry.getKey());
+                                }
+                                JsonObject field = fieldEntry.getValue().getAsJsonObject();
+                                if (field != null) {
+                                    if (Logger.isDebugEnabled()) {
+                                        Logger.log(className, methodName, Logger.LogType.DEBUG, "field JsonObject found");
+                                    }
+                                    // process values field
+                                    JsonArray valuesArray = field.getAsJsonArray("values");
+                                    if (valuesArray != null) {
+                                        if (Logger.isDebugEnabled()) {
+                                            Logger.log(className, methodName, Logger.LogType.DEBUG, "values array found");
+                                        }
+                                        try {
+                                            if (valuesArray.size() == 1 && valuesArray.get(0).getAsString().equals("${func.podlist()}")) {
+                                                ResolvedValue rv = context.resolve(valuesArray.get(0).getAsString());
+                                                if (Logger.isDebugEnabled()) {
+                                                    Logger.log(className, methodName, Logger.LogType.DEBUG, "values value: " + valuesArray.get(0).getAsString());
```

Comment:
Not sure if this was your intention, but this debug statement is essentially a static message. valuesArray.get(0).getAsString() always equals "${func.podlist()}".

Reply:
I have the log statement use the function for the potential case where the code handles all patterns.